llms/mlx_lm/models/phi3.py

```diff
@@ -73,7 +74,12 @@ def __call__(
         B, L, D = x.shape
 
         qkv = self.qkv_proj(x)
-        queries, keys, values = mx.split(qkv, 3, axis=-1)
+        if self.num_hidden_layers == 40:
+            query_pos = self.n_heads * self.head_dim
+            queries, keys, values = mx.split(qkv, [query_pos, query_pos + self.n_kv_heads * self.head_dim], axis=-1)
+        else:
+            queries, keys, values = mx.split(qkv, 3, axis=-1)
```

Comment:
I think you should just remove the condition on the number of layers and use the first split. It will give you the right thing when the `self.n_kv_heads == self.n_heads`.

Reply:
Read my mind!

It was a late this side to test it.

But I made the change this morning :)